rfc/replication.md

```diff
@@ -0,0 +1,125 @@
+
+The caveats for `space/blob/replicate` MUST include a `replicas` parameter, an unsigned integer indicating the number of copies the network should ensure are stored _in addition_ to the original.
+
+It is RECOMMENDED that the upload service mandate a minimum _and_ a maximum value. The minimum and maximum MAY be the same.
+
+e.g. `replicas: 2` will ensure 3 copies of the data exist in the network.
+
+The blob to be replicated and the location where the blob may be found are also required.
+
+```json5
+{
+  "iss": "did:key:zAlice",
+  "aud": "did:web:upload.service.example.com",
+  "att": [
+    {
+      "with": "did:key:zAliceSpace",
+      "can": "space/blob/replicate",
+      "nb": {
+        /** The blob that MUST be replicated. */
+        "blob": {
+          "digest": { "/": { "bytes": "..." } },
+          "size": 1234
+        },
+        /** Number of replicas to ensure. */
+        "replicas": 2,
+        /** A location commitment indicating where the blob MUST be fetched from. */
+        "location": { "/": "bafy..locationCommitment" }
+      }
+    }
+  ],
+  "prf": [],
+  "sig": "..."
+}
+```
+
+It is RECOMMENDED that the location commitment is included in the invocation.
+
+The receipt for `space/blob/replicate` includes effects (async tasks) for `blob/replica/transfer`. Successful completion of the `blob/replica/transfer` task indicates the replication target has transferred and stored the blob. The number of `blob/replica/transfer` tasks corresponds directly to number of replicas requested.
+
+Each replication task MUST target a _different_ storage node and they MUST NOT target the original upload target.
+
+The upload service MUST select storage node(s) and allocate replication space when the `space/blob/replicate` invocation is received. The upload service allocates replication space on storage nodes by issuing a `blob/replica/allocate` invocation:
+
+```json5
+{
+  "iss": "did:web:upload.service.example.com",
+  "aud": "did:key:zStorageNode",
+  "att": [
+    {
+      "with": "did:key:zStorageNode",
+      "can": "blob/replica/allocate",
+      "nb": {
+        /** The blob that was must be replicated. */
+        "blob": {
+          "digest": { "/": { "bytes": "..." } },
+          "size": 1234
+        },
+        /** DID of the space the blob has been allocated to. */
+        "space": { "/": { "bytes": "..." } },
+        /** A location commitment indicating where the blob MUST be fetched from. */
+        "location": { "/": "bafy..locationCommitment" },
+        /** The `space/blob/replicate` invocation that caused this allocation. */
+        "cause": { "/": "bafy..replicate" }
+      }
+    }
+  ],
+  "prf": [],
+  "sig": "..."
+}
+```
+
+The `blob/replica/allocate` task receipt includes an async task that will be performed by the storage node - `blob/replica/transfer`. The `blob/replica/transfer` task is completed when the storage node has transferred the blob from its location to the storage node.
```

Comment:
Similar question about receipts here, what goes into the receipts of `blob/replica/allocate` invocations, if anything? (aside from the mentioned async task)

Reply:
I just updated the blob spec to reflect the current implementation but yes the same (or similar) to blob allocate receipt https://github.com/storacha/specs/blob/main/w3-blob.md#allocate-blob-receipt